docs/ref/models/expressions.txt

```diff
@@ -649,6 +666,185 @@ should avoid them if possible.
+        documentation of your database before using this expression.
+
+.. _window-frames:
+
+Frames
+~~~~~~
+
+Two classes are used to model a frame to distinguish between a range-based
+sequence of rows, and an ordinary sequence of rows. This will be explained with
+some examples. Both of them return SQL according to this template::
+
+    %(frametype)s BETWEEN %(start)s AND %(end)s
+
+.. class:: ValueRange(start=None, end=None)
+
+    .. attribute:: frametype
+
+        This attribute is set to ``RANGE``.
+
+.. class:: RowRange(start=None, end=None)
+
+    .. attribute:: frametype
+
+        This attribute is set to ``ROWS``.
+
+Frames are used for narrowing which rows that should be used for computing the
+result. To ease the understanding of frames, they shift from some start point
+to some specified end point. Frames can be used with and without partitions,
+but it is recommended to specify an ordering of the window. In a frame, a peer
+in a frame is a row with an equivalent value, or all rows in case of an absent
+ordering-clause.
+
+The default starting point for a frame is ``UNBOUNDED PRECEDING`` which is the
+first row of the partition. The end point is always explicitly included in the
+SQL generated by the ORM and is by default ``UNBOUNDED FOLLOWING``. The default
+frame includes all rows from the partition to the last row in the set.
+
+The accepted values for the parameters for ``start`` and ``end`` are either
+``None``, an integer, or zero. The convention for the input values is as
+follows. A negative integer for ``start`` results in ``N preceding``, and None
+yields ``UNBOUNDED PRECEDING``. For both ``start`` and ``end``, zero will
+return ``CURRENT ROW``. Positive integers are accepted for ``end``.
+
+There is a difference what ``CURRENT ROW`` includes, i.e., when specified in
+``ROWS`` mode, the frame starts or ends with the current row. When specified in
+``RANGE``, the frame starts or ends at the first or last peer according to the
+ordering-clause. Thus, ``RANGE CURRENT ROW`` evaluates the expression for rows
+which have the same value specified by the ordering. Because the template is
+specific, this may be specified with::
+
+    ValueRange(start=0, end=0)
+
+We will use a similar with the movies from before to illustrate the use of
+``RowRange``. A movie studio produces many years a year, within many different
+genres. In this example, we compare movies to the two previous and two next
+movies released. In case there are not enough for the comparison, the
+comparison is just performed on the movies in the set..
+
+.. code-block:: python
+
+      >>> from django.db.models import Avg, F, RowRange, Window
+      >>>
+      >>> Movie.objects.prefetch_related('studio').annotate(
+      >>>     avg_rating=Window(
+      >>>         expression=Avg('rating'),
+      >>>         partition_by=[F('studio'), F('genre')],
+      >>>         order_by=F('released__year').asc(),
```

Comment:
Seems like this would only sort movies by year... would month/day ordering be unspecified then? Doesn't seem like that's what we want.

Reply:
It should be `ExtractYear` (or `TruncYear`).